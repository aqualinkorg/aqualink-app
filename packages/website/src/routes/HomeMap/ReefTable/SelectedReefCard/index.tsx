/* eslint-disable no-nested-ternary */
import React from "react";
import { isNumber } from "lodash";
import {
  Box,
  Button,
  Card,
  CardMedia,
  CircularProgress,
  Grid,
  Typography,
  Hidden,
  useTheme,
  useMediaQuery,
} from "@material-ui/core";
import LaunchIcon from "@material-ui/icons/Launch";
import { Link } from "react-router-dom";

import { useSelector } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import { sortByDate } from "../../../../helpers/sortDailyData";
import { formatNumber } from "../../../../helpers/numberUtils";

import { degreeHeatingWeeksCalculator } from "../../../../helpers/degreeHeatingWeeks";
import {
  reefDetailsSelector,
  reefLoadingSelector,
} from "../../../../store/Reefs/selectedReefSlice";
import { getReefNameAndRegion } from "../../../../store/Reefs/helpers";
import { Reef } from "../../../../store/Reefs/types";
import Chart from "../../../../common/Chart";
import { surveyListSelector } from "../../../../store/Survey/surveyListSlice";
import { convertDailyDataToLocalTime } from "../../../../helpers/dates";
import Chip from "../../../../common/Chip";

const useStyles = makeStyles((theme) => ({
  cardWrapper: {
    minHeight: "18rem",
    [theme.breakpoints.down("md")]: {
      minHeight: "22rem",
    },
  },
  mobileCardWrapperWithImage: {
    [theme.breakpoints.down("xs")]: {
      height: "42rem",
    },
  },
  mobileCardWrapperWithNoImage: {
    [theme.breakpoints.down("xs")]: {
      height: "27rem",
    },
  },
  card: {
    [theme.breakpoints.down("xs")]: {
      padding: 10,
    },
    padding: 20,
  },
  cardImage: {
    borderRadius: "4px 0 0 4px",
    height: "100%",

    [theme.breakpoints.down("xs")]: {
      height: 300,
    },
  },
  metricsContainer: {
    display: "flex",
    justifyContent: "space-around",
    alignItems: "center",
    flexDirection: "row",
    width: "100%",
    [theme.breakpoints.down("xs")]: {
      marginTop: theme.spacing(3),
      marginBottom: theme.spacing(3),
    },

    [theme.breakpoints.up("lg")]: {
      flexDirection: "column",
      alignItems: "start",
    },
  },
  launchIcon: {
    fontSize: 20,
    marginLeft: "0.5rem",
    color: "#2f2f2f",
    "&:hover": {
      color: "#2f2f2f",
    },
  },
  reefRegionName: {
    marginBottom: "0.6rem",
  },
  cardTitleWrapper: {
    maxWidth: "calc(100% - 88px)",
    marginRight: theme.spacing(1),
  },
  cardTitle: {
    width: "100%",
    display: "block",
    overflow: "hidden",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
  },
}));

type SelectedReefContentProps = {
  reef: Reef;
  imageUrl?: string | null;
};

const SelectedReefContent = ({ reef, imageUrl }: SelectedReefContentProps) => {
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
  const sortedDailyData = sortByDate(reef.dailyData, "date");
  const dailyDataLen = sortedDailyData.length;
  const { maxBottomTemperature, satelliteTemperature, degreeHeatingDays } =
    (dailyDataLen && sortedDailyData[dailyDataLen - 1]) || {};

  const metrics = [
    {
      label: "SURFACE TEMP",
      value: formatNumber(satelliteTemperature, 1),
      unit: " °C",
      display: isNumber(satelliteTemperature),
    },
    {
      label: "HEAT STRESS",
      value: formatNumber(degreeHeatingWeeksCalculator(degreeHeatingDays), 1),
      unit: " DHW",
      display: isNumber(degreeHeatingDays),
    },
    {
      label: `TEMP AT ${reef.depth}m`,
      value: formatNumber(maxBottomTemperature, 1),
      unit: " °C",
      display: isNumber(maxBottomTemperature),
    },
  ];

  const { name, region: regionName } = getReefNameAndRegion(reef);

  const ChartComponent = (
    <Chart
      reefId={reef.id}
      dailyData={convertDailyDataToLocalTime(reef.dailyData, reef.timezone)}
      surveys={[]}
      temperatureThreshold={
        reef.maxMonthlyMean ? reef.maxMonthlyMean + 1 : null
      }
      maxMonthlyMean={reef.maxMonthlyMean || null}
      background
    />
  );

  return (
    <Grid
      className={
        imageUrl
          ? `${classes.cardWrapper} ${classes.mobileCardWrapperWithImage}`
          : `${classes.cardWrapper} ${classes.mobileCardWrapperWithNoImage}`
      }
      container
      justify="space-between"
      spacing={1}
    >
      {imageUrl && (
        <Grid item xs={12} sm={6} lg={4}>
          <Box position="relative" height="100%">
            <Link to={`/reefs/${reef.id}`}>
              <CardMedia className={classes.cardImage} image={imageUrl} />
            </Link>

            <Hidden smUp>
              <Box
                bgcolor="rgba(3, 48, 66, 0.75)"
                height="40%"
                width="100%"
                position="absolute"
                top={0}
                left={0}
              >
                <Box position="absolute" top={16} left={16}>
                  <Typography variant="h5">{name}</Typography>

                  {regionName && (
                    <Typography variant="h6" style={{ fontWeight: 400 }}>
                      {regionName}
                    </Typography>
                  )}
                </Box>
              </Box>
            </Hidden>

            <Box position="absolute" bottom={16} px={2} width="100%">
              <Grid
                container
                alignItems="center"
                justify={
                  reef.videoStream && isMobile ? "space-between" : "flex-end"
                }
              >
                {reef.videoStream && isMobile && (
                  <Chip
                    live
                    liveText="LIVE VIDEO"
                    to={`/reefs/${reef.id}`}
                    width={80}
                  />
                )}
                <Grid item>
                  <Link
                    style={{ color: "inherit", textDecoration: "none" }}
                    to={`/reefs/${reef.id}`}
                  >
                    <Button size="small" variant="contained" color="primary">
                      EXPLORE
                    </Button>
                  </Link>
                </Grid>
              </Grid>
            </Box>
          </Box>
        </Grid>
      )}

      <Grid
        item
        xs={12}
        sm={imageUrl ? 6 : 12}
        lg={imageUrl ? 6 : 10}
        style={{ marginBottom: "2rem", maxHeight: "14rem" }}
      >
        <Box pb="0.5rem" pl="0.5rem" pt="1.5rem" fontWeight={400}>
          <Hidden xsDown={Boolean(imageUrl)}>
            <Grid container alignItems="center">
              <Grid item className={classes.cardTitleWrapper}>
                <Typography
                  className={classes.cardTitle}
                  color="textSecondary"
                  variant="h5"
                >
                  <span title={name || ""}>{name}</span>
                </Typography>
              </Grid>
              {reef.videoStream && (
                <Chip
                  live
                  liveText="LIVE VIDEO"
                  to={`/reefs/${reef.id}`}
                  width={80}
                />
              )}
            </Grid>
            <Typography
              color="textSecondary"
              variant="h6"
              className={`${classes.cardTitle} ${classes.reefRegionName}`}
            >
              <span title={regionName || ""}>{regionName}</span>
            </Typography>
          </Hidden>
          <Typography color="textSecondary" variant="caption">
            DAILY SURFACE TEMP. (°C)
          </Typography>
        </Box>
        <Hidden xsDown>
          <div>{ChartComponent}</div>
        </Hidden>
        <Hidden smUp>{ChartComponent}</Hidden>
      </Grid>

      <Grid
        item
        xs={12}
        lg={2}
        style={{
          display: "flex",
        }}
      >
        <div className={classes.metricsContainer}>
          {metrics.map(({ label, value, unit, display }) => (
            <div key={label}>
              {display && (
                <>
                  <Typography variant="caption" color="textSecondary">
                    {label}
                  </Typography>
                  <Typography variant="h4" color="primary">
                    {value}
                    &nbsp;
                    <Typography variant="h6" component="span">
                      {unit}
                    </Typography>
                  </Typography>
                </>
              )}
            </div>
          ))}
        </div>
      </Grid>
    </Grid>
  );
};

SelectedReefContent.defaultProps = {
  imageUrl: null,
};

const featuredReefId = process.env.REACT_APP_FEATURED_REEF_ID || "";

const SelectedReefCard = () => {
  const classes = useStyles();
  const reef = useSelector(reefDetailsSelector);
  const loading = useSelector(reefLoadingSelector);
  const surveyList = useSelector(surveyListSelector);

  const isFeatured = `${reef?.id}` === featuredReefId;

  const { featuredSurveyMedia } =
    sortByDate(surveyList, "diveDate", "desc").find(
      (survey) =>
        survey.featuredSurveyMedia &&
        survey.featuredSurveyMedia.type === "image"
    ) || {};

  const hasMedia = Boolean(featuredSurveyMedia?.url);

  return (
    <Box className={classes.card}>
      {!loading && (
        <Box mb={2}>
          <Typography variant="h5" color="textSecondary">
            {isFeatured ? "Featured Site" : "Selected Site"}
            {!hasMedia && (
              <Link to={`reefs/${reef?.id}`}>
                <LaunchIcon className={classes.launchIcon} />
              </Link>
            )}
          </Typography>
        </Box>
      )}

      <Card>
        {loading ? (
          <Box
            height="20rem"
            display="flex"
            alignItems="center"
            justifyContent="center"
            textAlign="center"
            p={4}
          >
            <CircularProgress size="3rem" thickness={1} />
          </Box>
        ) : reef ? (
          <SelectedReefContent
            reef={reef}
            imageUrl={featuredSurveyMedia?.url}
          />
        ) : null}
      </Card>
    </Box>
  );
};

export default SelectedReefCard;
