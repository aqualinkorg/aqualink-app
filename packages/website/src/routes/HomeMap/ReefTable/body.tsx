import {
  createStyles,
  Hidden,
  TableBody,
  TableCell,
  TableRow,
  Theme,
  Typography,
  useMediaQuery,
  useTheme,
  WithStyles,
  withStyles,
} from "@material-ui/core";
import ErrorIcon from "@material-ui/icons/Error";
import React, { useEffect, useState, useCallback } from "react";
import { useDispatch, useSelector } from "react-redux";
import { TableRow as Row } from "../../../store/Homepage/types";
import { constructTableData } from "../../../store/Reefs/helpers";
import { colors } from "../../../layout/App/theme";
import { dhwColorFinder } from "../../../helpers/degreeHeatingWeeks";
import { formatNumber } from "../../../helpers/numberUtils";
import { reefsToDisplayListSelector } from "../../../store/Reefs/reefsListSlice";
import {
  reefOnMapSelector,
  setReefOnMap,
  setSearchResult,
} from "../../../store/Homepage/homepageSlice";
import { getComparator, Order, OrderKeys, stableSort } from "./utils";
import { alertColorFinder } from "../../../helpers/bleachingAlertIntervals";
import { Collection } from "../../Dashboard/collection";

const SCROLLT_TIMEOUT = 500;

const RowNameCell = ({
  reef: { locationName, region },
  classes,
}: {
  reef: Row;
  classes: ReefTableBodyProps["classes"];
}) => {
  return (
    <TableCell className={classes.nameCells}>
      <Typography align="left" variant="h6" color="textSecondary">
        {locationName}
      </Typography>

      {locationName !== region && region && (
        <Typography className={classes.regionName} variant="subtitle1">
          {region}
        </Typography>
      )}
    </TableCell>
  );
};

const RowNumberCell = ({
  color,
  unit,
  decimalPlaces,
  value,
  classes,
  isExtended,
}: {
  color?: string;
  unit?: string;
  value: number | null;
  decimalPlaces?: number;
  classes: ReefTableBodyProps["classes"];
  isExtended?: boolean;
}) => {
  return (
    <TableCell
      className={
        isExtended ? classes.cellTextAlignExtended : classes.cellTextAlign
      }
    >
      <Typography
        variant="h6"
        style={{ color }}
        className={classes.numberCellsTitle}
      >
        {formatNumber(value, decimalPlaces)}
        <Hidden smUp>
          &nbsp;
          <Typography variant="h6" component="span">
            {unit}
          </Typography>
        </Hidden>
      </Typography>
    </TableCell>
  );
};

const RowAlertCell = ({
  reef: { alertLevel },
  classes,
}: {
  reef: Row;
  classes: ReefTableBodyProps["classes"];
}) => {
  return (
    <TableCell className={classes.cellTextAlign}>
      <ErrorIcon
        style={{
          color: alertColorFinder(alertLevel),
        }}
      />
    </TableCell>
  );
};

RowNumberCell.defaultProps = {
  unit: "",
  color: colors.black,
  decimalPlaces: 1,
  isExtended: false,
};

const ReefTableBody = ({
  order,
  orderBy,
  isExtended,
  collection,
  scrollTableOnSelection,
  scrollPageOnSelection,
  classes,
}: ReefTableBodyProps) => {
  const dispatch = useDispatch();
  const storedReefs = useSelector(reefsToDisplayListSelector);
  const reefsList = collection?.reefs || storedReefs || [];
  const reefOnMap = useSelector(reefOnMapSelector);
  const [selectedRow, setSelectedRow] = useState<number>();

  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.only("xs"));

  const mapElement = document.getElementById("sites-map");

  const handleClick = (event: unknown, reef: Row) => {
    setSelectedRow(reef.tableData.id);
    dispatch(setSearchResult());
    dispatch(setReefOnMap(reefsList[reef.tableData.id]));
    if (scrollPageOnSelection && mapElement) {
      mapElement.scrollIntoView({ block: "center", behavior: "smooth" });
    }
  };

  const scrollToRow = useCallback(
    (element: HTMLElement) =>
      setTimeout(
        () => element.scrollIntoView({ block: "center", behavior: "smooth" }),
        SCROLLT_TIMEOUT
      ),
    []
  );

  useEffect(() => {
    const index = reefsList.findIndex((item) => item.id === reefOnMap?.id);
    setSelectedRow(index);
  }, [reefOnMap, reefsList]);

  // scroll to the relevant reef row when reef is selected.
  useEffect(() => {
    const child = document.getElementById(`homepage-table-row-${selectedRow}`);
    // only scroll if not on mobile (info at the top is more useful than the reef row)
    if (child && !isMobile && scrollTableOnSelection) {
      scrollToRow(child);
    }
  }, [isMobile, scrollTableOnSelection, scrollToRow, selectedRow]);

  return (
    <TableBody>
      {stableSort<Row>(
        constructTableData(reefsList),
        getComparator(order, orderBy)
      ).map((reef) => {
        return (
          <TableRow
            id={`homepage-table-row-${reef.tableData.id}`}
            hover
            className={classes.tableRow}
            style={{
              backgroundColor:
                reef.tableData.id === selectedRow
                  ? colors.lighterBlue
                  : "white",
            }}
            onClick={(event) => handleClick(event, reef)}
            role="button"
            tabIndex={-1}
            key={reef.tableData.id}
          >
            <RowNameCell
              reef={reef}
              classes={{
                ...classes,
                nameCells: isExtended
                  ? classes.extendedTableNameCells
                  : classes.nameCells,
              }}
            />
            <RowNumberCell
              isExtended={isExtended}
              classes={classes}
              value={reef.sst}
              color={isExtended ? colors.black : colors.lightBlue}
              unit="°C"
            />
            {isExtended && (
              <RowNumberCell
                isExtended={isExtended}
                classes={classes}
                value={reef.historicMax}
                color={colors.black}
                unit="°C"
              />
            )}
            {isExtended && (
              <RowNumberCell
                isExtended={isExtended}
                classes={classes}
                value={reef.sstAnomaly}
                color={colors.black}
                unit="°C"
              />
            )}
            <RowNumberCell
              isExtended={isExtended}
              classes={classes}
              value={reef.dhw}
              color={dhwColorFinder(reef.dhw)}
              unit="DHW"
            />
            {isExtended && (
              <RowNumberCell
                isExtended={isExtended}
                classes={classes}
                value={reef.buoyTop}
                color={colors.black}
                unit="°C"
              />
            )}
            {isExtended && (
              <RowNumberCell
                isExtended={isExtended}
                classes={classes}
                value={reef.buoyBottom}
                color={colors.black}
                unit="°C"
              />
            )}
            <RowAlertCell reef={reef} classes={classes} />
          </TableRow>
        );
      })}
    </TableBody>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    nameCells: {
      paddingLeft: 10,
      maxWidth: 120,
      overflowWrap: "break-word",
      [theme.breakpoints.down("xs")]: { width: "35%", paddingRight: 0 },
    },
    extendedTableNameCells: {
      paddingLeft: 10,
      maxWidth: 200,
      overflowWrap: "break-word",
      [theme.breakpoints.down("xs")]: { width: "10%", paddingRight: 0 },
    },
    regionName: {
      color: "gray",
    },
    numberCellsTitle: {
      [theme.breakpoints.down("xs")]: { fontWeight: 600 },
    },
    cellTextAlign: {
      textAlign: "left",
      [theme.breakpoints.down("xs")]: {
        textAlign: "right",
      },
    },
    cellTextAlignExtended: {
      [theme.breakpoints.down("xs")]: {
        paddingLeft: 10,
        paddingRight: 0,
        textAlign: "left",
      },
    },
    tableRow: {
      cursor: "pointer",
      borderTop: `1px solid ${theme.palette.grey["300"]}`,
    },
  });

type ReefTableBodyIncomingProps = {
  order: Order;
  orderBy: OrderKeys;
  isExtended?: boolean;
  collection?: Collection;
  scrollTableOnSelection?: boolean;
  scrollPageOnSelection?: boolean;
};

ReefTableBody.defaultProps = {
  isExtended: false,
  collection: undefined,
  scrollTableOnSelection: true,
  scrollPageOnSelection: false,
};

type ReefTableBodyProps = WithStyles<typeof styles> &
  ReefTableBodyIncomingProps;

export default withStyles(styles)(ReefTableBody);
