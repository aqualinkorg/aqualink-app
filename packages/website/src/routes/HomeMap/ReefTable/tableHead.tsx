import React from "react";
import {
  TableHead,
  TableRow,
  TableCell,
  TableSortLabel,
  Typography,
  createStyles,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import type { Order, OrderKeys } from "./utils";

const ColumnTitle = ({ title, unit }: { title: string; unit?: string }) => (
  <Typography variant="h6" style={{ color: "black" }} noWrap>
    {title}
    {unit && (
      <Typography
        variant="subtitle2"
        style={{ color: "black" }}
        component="span"
      >{` (${unit})`}</Typography>
    )}
  </Typography>
);

ColumnTitle.defaultProps = {
  unit: undefined,
};

const EnhancedTableHead = (props: EnhancedTableProps) => {
  const createSortHandler = (property: OrderKeys) => (
    event: React.MouseEvent<unknown>
  ) => {
    props.onRequestSort(event, property);
  };

  const headCells: HeadCell[] = [
    {
      id: "locationName",
      label: "SITE",
      width: "40%",
    },
    {
      id: "temp",
      label: "TEMP",
      unit: "Â°C",
    },
    {
      id: "dhw",
      label: "STRESS",
      unit: "DHW",
    },
    {
      id: "alert",
      label: "ALERT",
      width: "5%",
    },
  ];

  return (
    <TableHead style={{ backgroundColor: "rgb(244, 244, 244)" }}>
      <TableRow>
        {headCells.map((headCell) => (
          <TableCell
            className={props.classes.headCells}
            key={headCell.id}
            style={{ width: headCell.width }}
            align="left"
            padding="default"
            sortDirection={props.orderBy === headCell.id ? props.order : false}
          >
            <TableSortLabel
              active={props.orderBy === headCell.id}
              direction={props.orderBy === headCell.id ? props.order : "asc"}
              onClick={createSortHandler(headCell.id)}
            >
              <ColumnTitle title={headCell.label} unit={headCell.unit} />
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

interface HeadCell {
  id: OrderKeys;
  label: string;
  unit?: string;
  width?: string;
}

interface EnhancedTableIncomingProps {
  onRequestSort: (
    event: React.MouseEvent<unknown>,
    property: OrderKeys
  ) => void;
  order: Order;
  orderBy: OrderKeys;
}

const styles = () =>
  createStyles({
    headCells: {
      paddingRight: 0,
      paddingLeft: 10,
    },
  });

type EnhancedTableProps = WithStyles<typeof styles> &
  EnhancedTableIncomingProps;

export default withStyles(styles)(EnhancedTableHead);
