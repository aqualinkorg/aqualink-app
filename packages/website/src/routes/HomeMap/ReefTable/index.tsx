import React, { useState } from "react";
import {
  Box,
  CircularProgress,
  createStyles,
  Grid,
  Hidden,
  IconButton,
  Table,
  TableContainer,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import ArrowDownwardIcon from "@material-ui/icons/ArrowDownward";

import { useSelector } from "react-redux";
import SelectedReefCard from "./SelectedReefCard";
import ReefTableBody from "./body";
import { Order, OrderKeys } from "./utils";
import { reefsListLoadingSelector } from "../../../store/Reefs/reefsListSlice";
import EnhancedTableHead from "./tableHead";
import { useWindowSize } from "../../../helpers/useWindowSize";

const SMALL_HEIGHT = 720;
const SMALL_WIDTH = 600;

const ReefTable = ({ openDrawer, classes }: ReefTableProps) => {
  const loading = useSelector(reefsListLoadingSelector);
  const { height, width } = useWindowSize() || {};

  const [order, setOrder] = useState<Order>("desc");
  const [orderBy, setOrderBy] = useState<OrderKeys>("alert");

  const handleRequestSort = (event: unknown, property: OrderKeys) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const showTable = (width && width >= SMALL_WIDTH) || openDrawer;

  return (
    <>
      <Hidden smUp>
        <Grid container justify="center" style={{ marginBottom: "-3rem" }}>
          <Grid item>
            <IconButton>
              {openDrawer ? <ArrowDownwardIcon /> : <ArrowUpwardIcon />}
            </IconButton>
          </Grid>
        </Grid>
        {openDrawer ? null : (
          <Typography
            style={{
              position: "relative",
              margin: "1rem 0 0.5rem 1rem",
            }}
            variant="h5"
            color="textSecondary"
          >
            All Reefs
          </Typography>
        )}
      </Hidden>
      {showTable && (
        <>
          <SelectedReefCard />
          <Box
            className={
              height && height > SMALL_HEIGHT
                ? `${classes.tableHolder} ${classes.scrollable}`
                : `${classes.tableHolder}`
            }
            display="flex"
            flexDirection="column"
            flex={1}
          >
            <TableContainer>
              <Table stickyHeader className={classes.table}>
                <Hidden xsDown>
                  <EnhancedTableHead
                    order={order}
                    orderBy={orderBy}
                    onRequestSort={handleRequestSort}
                  />
                </Hidden>
                <ReefTableBody order={order} orderBy={orderBy} />
              </Table>
            </TableContainer>
            {loading && (
              <Box
                display="flex"
                flex={1}
                alignItems="center"
                justifyContent="center"
              >
                <CircularProgress size="10rem" thickness={1} />
              </Box>
            )}
          </Box>
        </>
      )}
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    tableHolder: {
      paddingLeft: 10,
      [theme.breakpoints.down("xs")]: {
        paddingLeft: 0,
        height: "auto",
      },
    },
    scrollable: {
      overflowY: "auto",
    },
    table: {
      [theme.breakpoints.down("xs")]: {
        tableLayout: "fixed",
      },
    },
  });

interface ReefTableIncomingProps {
  openDrawer: boolean;
}

type ReefTableProps = ReefTableIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(ReefTable);
