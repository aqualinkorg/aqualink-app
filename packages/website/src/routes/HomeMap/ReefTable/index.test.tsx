import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import ReefTable from ".";
import { mockReef } from "../../../mocks/mockReef";

jest.mock("./SelectedReefCard", () => "Mock-SelectedReefCard");

const mockStore = configureStore([]);

describe("ReefTable", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      reefsList: {
        list: [mockReef],
        loading: false,
        error: null,
      },
      selectedReef: {
        loading: false,
        error: null,
      },
      homepage: {
        reefOnMap: null,
      },
    });

    const openDrawer = false;

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <ReefTable openDrawer={openDrawer} />
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
