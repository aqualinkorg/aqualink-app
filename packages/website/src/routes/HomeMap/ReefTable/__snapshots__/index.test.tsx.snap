// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReefTable should render with given state from Redux store 1`] = `
<div>
  <mock-hidden
    smup="true"
  >
    <mock-box
      display="flex"
      justifycontent="center"
      marginbottom="3"
      margintop="2"
      width="100vw"
    >
      <mock-box
        classname="ReefTable-topHandle-5"
      />
      <mock-typography
        classname="ReefTable-allReefsText-7"
        color="textSecondary"
        variant="h5"
      >
        All Reefs
      </mock-typography>
    </mock-box>
  </mock-hidden>
  <mock-selectedreefcard />
  <mock-hidden
    smup="true"
  >
    <mock-box
      alignitems="center"
      display="flex"
      paddingx="2"
      paddingy="3"
    >
      <mock-typography
        variant="h5"
      >
        Sort By: 
      </mock-typography>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline ReefTable-mobileSortSelect-6"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          <mock-typography
            color="primary"
            variant="h4"
          >
            Alert
              
            <arrowupward
              font-size="small"
            />
          </mock-typography>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="alert-desc"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <mock-menu
          anchorel="[object HTMLDivElement]"
          id="menu-"
          menulistprops="[object Object]"
          open="false"
          paperprops="[object Object]"
        >
          <mock-menuitem
            data-value="alert-asc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Alert
                
              <arrowdownward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="true"
            data-value="alert-desc"
            role="option"
            selected="true"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Alert
                
              <arrowupward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="depth-asc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Depth
                
              <arrowdownward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="depth-desc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Depth
                
              <arrowupward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="dhw-asc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              DHW
                
              <arrowdownward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="dhw-desc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              DHW
                
              <arrowupward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="locationName-asc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Location Name
                
              <arrowdownward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="locationName-desc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Location Name
                
              <arrowupward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="temp-asc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Temp
                
              <arrowdownward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            data-value="temp-desc"
            role="option"
            selected="false"
          >
            <mock-typography
              color="primary"
              variant="h4"
            >
              Temp
                
              <arrowupward
                font-size="small"
              />
            </mock-typography>
          </mock-menuitem>
        </mock-menu>
      </div>
    </mock-box>
  </mock-hidden>
  <mock-box
    classname="ReefTable-tableHolder-1 ReefTable-scrollable-2"
    display="flex"
    flex="1"
    flexdirection="column"
  >
    <mock-tablecontainer>
      <mock-table
        classname="ReefTable-table-3"
        stickyheader="true"
      >
        <mock-hidden
          xsdown="true"
        >
          <mock-tablehead
            style="background-color: rgb(244, 244, 244);"
          >
            <mock-tablerow>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-10"
                padding="default"
                sortdirection="false"
                style="width: 40%;"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    SITE
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-10"
                padding="default"
                sortdirection="false"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    SST
                    <mock-typography
                      component="span"
                      style="color: black;"
                      variant="subtitle2"
                    >
                       (Â°C)
                    </mock-typography>
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-10"
                padding="default"
                sortdirection="false"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    STRESS
                    <mock-typography
                      component="span"
                      style="color: black;"
                      variant="subtitle2"
                    >
                       (DHW)
                    </mock-typography>
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-10"
                padding="default"
                sortdirection="desc"
                style="width: 5%;"
              >
                <mock-tablesortlabel
                  active="true"
                  direction="desc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    ALERT
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
            </mock-tablerow>
          </mock-tablehead>
        </mock-hidden>
        <mock-tablebody>
          <mock-tablerow
            classname="ReefTableBody-tableRow-15"
            hover="true"
            id="homepage-table-row-0"
            role="button"
            style="background-color: white;"
            tabindex="-1"
          >
            <mock-tablecell
              classname="ReefTableBody-nameCells-11"
            >
              <mock-typography
                align="left"
                color="textSecondary"
                variant="h6"
              >
                Hawaii
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="ReefTableBody-cellTextAlign-14"
            >
              <mock-typography
                classname="ReefTableBody-numberCellsTitle-13"
                style="color: rgb(22, 141, 189);"
                variant="h6"
              >
                23.0
                <mock-hidden
                  smup="true"
                >
                  Â 
                  <mock-typography
                    component="span"
                    variant="h6"
                  >
                    Â°C
                  </mock-typography>
                </mock-hidden>
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="ReefTableBody-cellTextAlign-14"
            >
              <mock-typography
                classname="ReefTableBody-numberCellsTitle-13"
                style="color: rgb(117, 180, 114);"
                variant="h6"
              >
                4.9
                <mock-hidden
                  smup="true"
                >
                  Â 
                  <mock-typography
                    component="span"
                    variant="h6"
                  >
                    DHW
                  </mock-typography>
                </mock-hidden>
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="ReefTableBody-cellTextAlign-14"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(198, 229, 250);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
                />
              </svg>
            </mock-tablecell>
          </mock-tablerow>
        </mock-tablebody>
      </mock-table>
    </mock-tablecontainer>
  </mock-box>
</div>
`;
